recipes/moc/all/conanfile.py

```diff
@@ -0,0 +1,91 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from os import rename
+from os.path import isdir, join
+from shutil import move
+from glob import glob
+
+class MocConan(ConanFile):
+    name = "moc"
+    homepage = "https://www.github.com/zuut/moc"
+    description = "Moc, the marked-up object compiler"
+    topics = ("conan", "moc", "idl", "code generation")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"    
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = { "fPIC": [True, False] }
+    default_options = { "fPIC": True }
+    _cmake = None
+    _supported_compilers = [
+        ("Linux", "gcc", "6"),
+        ("Linux", "clang", "6")
+    ]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if not self._check_compiler():
+            raise ConanInvalidConfiguration("%s package is not compatible with os %s and compiler %s version %s." % (self.name, self.settings.os, self.settings.compiler, self.settings.compiler.version))
+
+    def _check_compiler(self):
+        os = self.settings.os
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(compiler.version)
+        return any(os == sc[0] and compiler == sc[1] and compiler_version >= sc[2] for sc in self._supported_compilers)
+
+    def requirements(self):
+        self.requires("flex/2.6.4")
+        self.requires("bison/3.7.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(source_folder=self._source_subfolder)
```

Comment:
I always worry when I see this since it skips our cmake wrapper... I did not read the comments in case this was discussed

Reply:
Can you tell me what wrapper you're referring to? This was used by the recastnavigation recipe which was mentioned in https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#how-to-provide-a-good-recipe . NOTE, that section also states that all the recipes in CCI are good examples. One thing that would be helpful is to update the doc and provide a good example of what you want instead of waiting until review time. It would also make the life of new contributors like myself much easier.

Reply:
In reference to this one line... the examples are correct

https://github.com/conan-io/conan-center-index/blob/9f32bf3f48411fd91680ca0190317b3a7126f8b1/recipes/recastnavigation/all/conanfile.py#L52

https://github.com/conan-io/conan-center-index/blob/9f32bf3f48411fd91680ca0190317b3a7126f8b1/recipes/szip/all/conanfile.py#L66

You just set the wrong variable... which is why we do code reviews

Reply:
Please follow @prince-chrismc suggestion, CMake wrapper is important to properly inject settings from profile.

Do not resolve reviews so quickly please, it makes life of reviewers harder.